components/datadog/kubernetesagentparams/params.go

```diff
@@ -0,0 +1,132 @@
+package kubernetesagentparams
+
+import (
+	"fmt"
+
+	"github.com/DataDog/test-infra-definitions/common"
+	"github.com/DataDog/test-infra-definitions/common/config"
+	"github.com/DataDog/test-infra-definitions/common/utils"
+	"github.com/DataDog/test-infra-definitions/components/datadog/fakeintake"
+
+	"github.com/pulumi/pulumi/sdk/v3/go/pulumi"
+)
+
+const (
+	defaultAgentNamespace = "datadog"
+)
+
+// Params defines the parameters for the Kubernetes Agent installation.
+// The Params configuration uses the [Functional options pattern].
+//
+// The available options are:
+//   - [WithImageTag]
+//   - [WithRepository]
+//   - [WithFullImagePath]
+//   - [WithPulumiDependsOn]
+//   - [WithDeployWindows]
+//   - [WithHelmValues]
+//   - [WithNamespace]
+//   - [WithHostName]
+//	 - [WithFakeintake]
+//
+// [Functional options pattern]: https://dave.cheney.net/2014/10/17/functional-options-for-friendly-apis
+
+type Params struct {
+	// FullImagePath is the full path of the docker agent image to use.
+	// It has priority over ImageTag and Repository.
+	FullImagePath string
+	// ImageTag is the docker agent image tag to use.
+	ImageTag string
+	// Repository is the docker repository to use.
+	Repository string
+	// Namespace is the namespace to deploy the agent to.
+	Namespace string
+	// HelmValues is the Helm values to use for the agent installation.
+	HelmValues pulumi.AssetOrArchiveArray
+	// PulumiDependsOn is a list of resources to depend on.
+	PulumiDependsOn []pulumi.ResourceOption
+	// FakeIntake is the fake intake to use for the agent installation.
+	FakeIntake *fakeintake.Fakeintake
+	// DeployWindows is a flag to deploy the agent on Windows.
+	DeployWindows bool
+}
+
+type Option = func(*Params) error
+
+func NewParams(e *config.CommonEnvironment, options ...Option) (*Params, error) {
+	version := &Params{
+		Namespace: defaultAgentNamespace,
+	}
+
+	if e.PipelineID() != "" && e.CommitSHA() != "" {
+		options = append(options, WithFullImagePath(utils.BuildDockerImagePath("669783387624.dkr.ecr.us-east-1.amazonaws.com/agent", fmt.Sprintf("%s-%s", e.PipelineID(), e.CommitSHA()))))
```

Comment:
That is leaking two abstractions we don't want to leak: AWS + ECR.

Reply:
I removed it from the `NewParams`. To avoid leakage I think the solution is to compute the image on the Provider side (on datadog-agent) and to pass a `WithFullImagePath` option with the computed image

Reply:
Updated to use the new cross cloud provider parameters introduced in https://github.com/DataDog/test-infra-definitions/pull/632